src/login/msal/MsalAuthProvider.ts

```diff
@@ -37,10 +37,6 @@ export class MsalAuthProvider extends AuthProviderBase<AuthenticationResult> {
 		this.publicClientApp = new PublicClientApplication(msalConfiguration);
 	}
 
-	public async loginWithoutLocalServer(_clientId: string, _environment: Environment, _isAdfs: boolean, _tenantId: string): Promise<AuthenticationResult> {
```

Comment:
Bit hard to tell - was this PR just moving code around or where there also changes specifically to make it work with msal?

Reply:
There were a few changes in addition to moving code:

### `loginWithoutLocalServer`
```diff
-	public async loginWithoutLocalServer(clientId: string, environment: Environment, isAdfs: boolean, tenantId: string): Promise<TokenResponse[]> {	
+	public async loginWithoutLocalServer(clientId: string, environment: Environment, isAdfs: boolean, tenantId: string): Promise<TLoginResult> {
		const callbackUri: Uri = await env.asExternalUri(Uri.parse(`${env.uriScheme}://ms-vscode.azure-account`));	
		const nonce: string = randomBytes(16).toString('base64');	
		const port: string | number = (callbackUri.authority.match(/:([0-9]*)$/) || [])[1] || (callbackUri.scheme === 'https' ? 443 : 80);	
		const callbackEnvironment: string = getCallbackEnvironment(callbackUri);	
		const state: string = `${callbackEnvironment}${port},${encodeURIComponent(nonce)},${encodeURIComponent(callbackUri.query)}`;	
		const signInUrl: string = `${environment.activeDirectoryEndpointUrl}${isAdfs ? '' : `${tenantId}/`}oauth2/authorize`;	
		let uri: Uri = Uri.parse(signInUrl);	
		uri = uri.with({	
			query: `response_type=code&client_id=${encodeURIComponent(clientId)}&redirect_uri=${redirectUrlAAD}&state=${state}&resource=${environment.activeDirectoryResourceId}&prompt=select_account`	
		});	
		void env.openExternal(uri);	
-		const timeoutPromise = new Promise((_resolve: (value: TokenResponse) => void, reject) => {	
+		const timeoutPromise = new Promise((_resolve: (value: TLoginResult) => void, reject) => {
			const wait = setTimeout(() => {	
				clearTimeout(wait);	
				reject('Login timed out.');	
			}, 1000 * 60 * 5)	
		});	
-		const tokenResponse: TokenResponse = await Promise.race([this.exchangeCodeForToken(clientId, environment, tenantId, redirectUrlAAD, state), timeoutPromise]);	
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion	
-		await storeRefreshToken(environment, tokenResponse.refreshToken!);	
-		return getTokensFromToken(environment, tenantId, tokenResponse);	
+		return await Promise.race([this.exchangeCodeForToken(clientId, environment, tenantId, redirectUrlAAD, state), timeoutPromise]);
	}	

```

### `exchangeCodeForToken`
```diff
-	private async exchangeCodeForToken(clientId: string, environment: Environment, tenantId: string, callbackUri: string, state: string): Promise<TokenResponse> {
+	private async exchangeCodeForToken(clientId: string, environment: Environment, tenantId: string, callbackUri: string, state: string): Promise<TLoginResult> {
		let uriEventListener: Disposable;
-		return new Promise((resolve: (value: TokenResponse) => void , reject) => {
+ 		return new Promise((resolve: (value: TLoginResult) => void , reject) => {
			uriEventListener = this.handler.event(async (uri: Uri) => {
				try {
					/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access */
					const query = parseQuery(uri);
					const code = query.code;

					// Workaround double encoding issues of state
					if (query.state !== state && decodeURIComponent(query.state) !== state) {
						throw new Error('State does not match.');
					}
					/* eslint-enable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access */

-					resolve(await getTokenWithAuthorizationCode(clientId, environment, callbackUri, tenantId, code));
+					resolve(await this.loginWithAuthCode(code, callbackUri, clientId, environment, tenantId));
				} catch (err) {
					reject(err);
				}
			});
		}).then(result => {
			uriEventListener.dispose()
			return result;
		}).catch(err => {
			uriEventListener.dispose();
			throw err;
		});
	}
```